src/cloud/iotile-cloud-serv.ts

```diff
@@ -357,11 +357,10 @@ export class IOTileCloud {
    *
    * @returns {Device[]} A list of the IOTile devices.
    */
-  // TODO: get by pages of 500
   public async fetchAllDevices() {
     let that = this;
     return new Promise<Device[]>(function(resolve, reject) {
-      that.fetchFromServer('/device/?page_size=1000') //The default limit of devices returned per page is apparently 100
+      that.fetchPagesFromServer('/device/', 500)
```

Comment:
You may want to look into https://github.com/iotile/ng-iotile-cloud/blob/master/src/models/apifilter.ts and enhance the fetchPageFromService not to take 500, but to take an apiFilter, allowing you to reuse this function to filter a lot more. You can see how the apiFilter is use as an optional argument e.g. https://github.com/iotile/ng-iotile-cloud/blob/master/src/cloud.service.ts#L254